pkg/controllermanager/controller/seed/seed_control.go

```diff
@@ -227,13 +228,18 @@ func (c *defaultControl) ReconcileSeed(obj *gardencorev1alpha1.Seed, key string)
 			return nil
 		}
 
-		parentLogMessage := "Can't delete Seed, because the following objects are still referencing it: "
+		parentLogMessage := "Can't delete Seed, because the following objects are still referencing it:"
 		if len(associatedShoots) != 0 {
-			seedLogger.Infof("%s Shoots=%v", parentLogMessage, associatedShoots)
+			message := fmt.Sprintf("%s Shoots=%v", parentLogMessage, associatedShoots)
+			seedLogger.Info(message)
+			c.recorder.Event(seed, corev1.EventTypeNormal, v1alpha1constants.EventResourceReferenced, message)
 		}
 		if len(associatedBackupBuckets) != 0 {
-			seedLogger.Infof("%s BackupBuckets=%v", parentLogMessage, associatedBackupBuckets)
+			message := fmt.Sprintf("%s BackupBuckets=%v", parentLogMessage, associatedBackupBuckets)
+			seedLogger.Info(message)
+			c.recorder.Event(seed, corev1.EventTypeNormal, v1alpha1constants.EventResourceReferenced, message)
```

Comment:
Instead of sending two log messages and two events here, can we first collect and construct the message that we wanna send and then log+record it only once?

Reply:
I actually thought about that, too. The reason why I reused the current structure with two logs is because of the deduplication mechanism in the Events API. As long as the related object, event type, reason and most importantly message are the same, events are aggregated into one event series. As soon as one of them (e.g. event message) changes, it will be displayed to the user as a seperate event (series). This means that, when one of the referencing objects is indeed deleted, it will disappear from the message and therefore a new event series will be displayed to the user, for example:

```
Events:
  Type    Reason              Age                  From                         Message
  ----    ------              ----                 ----                         -------
  Normal  ResourceReferenced  97s (x15 over 3m7s)  gardener-controller-manager  ControllerInstallations for seed "dev-aws" still pending, cannot release seed
  Normal  ResourceReferenced  24s (x9 over 2m23s)  gardener-controller-manager  Can't delete Seed, because the following objects are still referencing it: Shoots=[garden-dev/fake-aws-1 garden-dev/fake-aws-2] BackupBuckets=[5c7171fe-abcd-11e9-9dcd-acde48001122]
  Normal  ResourceReferenced  15s                  gardener-controller-manager  Can't delete Seed, because the following objects are still referencing it: Shoots=[garden-dev/fake-aws-1 garden-dev/fake-aws-2]
```

I think, this might be confusing to the user. Yes, the problem is actually the same, if we split it into seperate events per resource, but at least it is a little less confusing and consistent with my other changes in the deletion flow of resources that can only be referenced by one other resource type.

Actually, I am not sure, what is the right way to do it in this case.
Just wanted to give some context of my considerations. WDYT now?

Reply:
👍 okay!